src/components/layout/TransactionWorkflow/TransactionWorkflow.tsx

```diff
@@ -98,6 +99,26 @@ function TransactionWorkflow({
         initialState: payload.initialState,
         tags: payload?.tags,
       });
+    } else if (
+      interactionType === INTERACTION_TYPES.BUY_RECORD &&
+      isObjectOfTransactionPayloadType<BuyRecordPayload>(
+        payload,
+        TRANSACTION_DATA_KEYS[INTERACTION_TYPES.BUY_RECORD].keys,
+      ) &&
+      payload.contractTxId === 'atomic' &&
+      payload.state
+    ) {
+      const writeInteractionId = await arweaveDataProvider.registerAtomicName({
```

Comment:
how we deal with a failure on `registerAtomicName`?

Reply:
this is an operation performed in `deployTransaction` - for each of the conditions, we assign the resulting transansaction id (writeInteractionId) to the locally scoped var `originalTxId` which can be undefined, if undefined we throw an error.

Reply:
https://github.com/ar-io/pdns-react/pull/241/commits/62cf09030fe95ab9c636d80160c0353354013077